File: 2-2/Sistem Analizi ve Tasarımı/projeler/2025/mert_guler/backend/src/main/java/com/savt/listopia/model/movie/Movie.java
package com.savt.listopia.model.movie;


import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.savt.listopia.model.movie.image.MovieImage;
import jakarta.annotation.Nullable;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "movies")
public class Movie {
    @Id
    private Integer movieId;

    private String originalLanguage;

    private String originalTitle;

    private String title;

    @Column(columnDefinition = "TEXT", length = 2048)
    private String overview;

    @Column(columnDefinition = "TEXT", length = 2048)
    private String tagline;

    private Double popularity;

    private String releaseDate;

    private String trailerKey;

    private String trailerLink;

    private Integer runtime;

    private Integer watchCount = 0;

    private Integer likeCount = 0;

    private Long clickCount = 0L;

    @NotNull
    private String imdbId;

    @Nullable
    private String wikidataId;

    @Nullable
    private String facebookId;

    @Nullable
    private String instagramId;

    @Nullable
    private String twitterId;

    @OneToMany(mappedBy = "movie", cascade = {CascadeType.MERGE, CascadeType.PERSIST}, orphanRemoval = true)
    @JsonManagedReference
    private List<MovieImage> backdrops;

    @JsonManagedReference
    @OneToMany(mappedBy = "movie", cascade = {CascadeType.MERGE, CascadeType.PERSIST}, orphanRemoval = true)
    private List<MovieImage> logos;

    @JsonManagedReference
    @OneToMany(mappedBy = "movie", cascade = {CascadeType.MERGE, CascadeType.PERSIST}, orphanRemoval = true)
    private List<MovieImage> posters;

    @JsonManagedReference
    @OneToMany(mappedBy = "movie", cascade = {CascadeType.MERGE, CascadeType.PERSIST}, orphanRemoval = true)
    private List<MovieCast> movieCasts = new ArrayList<>();

    @JsonManagedReference
    @OneToMany(mappedBy = "movie", cascade = {CascadeType.MERGE, CascadeType.PERSIST}, orphanRemoval = true)
    private List<MovieCrew> movieCrews = new ArrayList<>();

    @ManyToMany
    @JoinTable(name = "movie_genres",
            joinColumns = @JoinColumn(name = "movie_id"),
            inverseJoinColumns = @JoinColumn(name = "genre_id"))
    private List<Genre> genres;
}

File: 2-2/Sistem Analizi ve Tasarımı/projeler/2025/mert_guler/frontend/src/types/movie.ts
import { Genre } from './genre';
import { User } from './user';

export interface Movie {
    movieId: number;
    originalLanguage: string;
    originalTitle: string;
    title: string;
    overview: string;
    tagline: string;
    releaseDate: string;
    trailerLink: string;
    runtime: number;
    backdrop: string;
    poster: string;
    logo: string | null;
    genres: Genre[];
    watchCount: number;
    likeCount: number;
    ratingCount: number;
    imdbId: string;
    wikidataId: string;
    facebookId: string | null;
    instagramId: string | null;
    twitterId: string | null;
    clickCount: number | null;
}

export interface Comment {
    commentId: number;   // ID of the comment
    movieId: number;     // ID of the movie the comment is related to
    user: User;
    sentAt: number;      // Timestamp when the comment was sent (in milliseconds)
    message: string;     // The actual comment message
    isSpoiler: boolean;  // Flag to indicate if the comment contains spoilers
    isUpdated: boolean;  // Flag to indicate if the comment was updated
}


